dracut.sh

```diff
@@ -2619,12 +2619,50 @@ if [[ $uefi == yes ]]; then
         unset uefi_splash_image
     fi
 
+    if [[ -n ${uefi_tpm2_pcr_sig_private_key} && -n ${uefi_tpm2_pcr_sig_public_key} ]]; then
+        if command -v systemd-measure &> /dev/null; then
+            systemd_measure="systemd-measure"
+        else
+            if [[ -x "${dracutsysrootdir}${systemdutildir}/systemd-measure" ]]; then
+                systemd_measure="${dracutsysrootdir}${systemdutildir}/systemd-measure"
+            else
+                dfatal "systemd-measure not found"
+                exit 1
+            fi
+        fi
+
+        dinfo "*** Computing systemd-measure TPM2 PCR signature ***"
+
+        measure_parameters="--initrd=${DRACUT_TMPDIR}/initramfs.img \
+            --linux="$kernel_image" \
+            ${uefi_osrelease:+--osrel="$uefi_osrelease"} \
+            ${uefi_cmdline:+--cmdline="$uefi_cmdline"} \
+            ${uefi_splash_image:+--splash="$uefi_splash_image"} \
+            --pcrpkey=${uefi_tpm2_pcr_sig_public_key} \
+            --bank=sha1 \
+            --bank=sha256 \
```

Comment:
Hard coding sha1 and sha256 here is suboptimal. If your TPM supports a newer hash type, this will force the usage of a less secure version.
I think I get why you do it like this: If I specify a newer hash type (or none at all and rely the default, which is all hash types) on my system when using the `sign` verb, it will error out, since my TPM doesn't support it AFAICT.
So this is arguably an upstream issue, since I also don't know of an elegant way to solve this.
But this doesn't change the result: So maybe also add an config option for this too?

Reply:
I think adding a config option with a default value of `sha1 sha256` or something like that would be the way to go. thanks!